app/controllers/pets_controller.rb

```diff
@@ -49,4 +33,32 @@ def destroy
   def pet_params
     params.permit(:id, :name, :age, :breed, :adoptable, :shelter_id)
   end
+
+  def index_filter
+    if params[:search].present?
+      @pets = Pet.search(params[:search])
+    else
+      @pets = Pet.adoptable
+    end
+  end
+
+  def update_filter
+    pet = Pet.find(params[:id])
+    if pet.update(pet_params)
+      redirect_to "/pets/#{pet.id}"
+    else
+      redirect_to "/pets/#{pet.id}/edit"
+      flash[:alert] = "Error: #{error_message(pet.errors)}"
+    end
+  end
+
+  def create_filter
+    pet = Pet.new(pet_params)
+    if pet.save
```

Comment:
If you're interested in trying to get rid of the `if/else` condition for error handling in the controller action (or if that conditional complexity is what makes you want to create a filter), another option is trying to build a global error handling method in the ApplicationController. This is something we'll see more in Mod 3, but take a look at this article and you could consider trying to create a parent method that rescues from ActiveRecord::RecordInvalid errors! https://www.martin-brennan.com/rails-global-rescue-from-error-application-controller/

Reply:
This is just an alternative and possible optimization that we haven't learned about at all yet, but it might be fun for you to check out!